action.yml

```
@@ -1,139 +1,161 @@
 name: 'Setup GAP for package testing'
-description: 'Download and compile CI scripts, GAP and its packages'
+description: 'Download and compile GAP and its packages'
 inputs:
-  GAP_PKGS_TO_CLONE:
-    description: 'the GAP packages to clone'
+  gap-version:
+    description: 'The GAP version or branch to build'
     required: false
-    default: ''
-  GAP_PKGS_TO_BUILD:
-    description: 'the GAP packages to build'
-    required: false
-    default: 'io profiling'
-  GAPBRANCH:
-    description: 'the gap branch to clone'
+    default: 'latest'
+  repository:
+    description: 'The GitHub repository from which to clone GAP'
     required: false
-    default: 'master'
-  HPCGAP:
-    description: 'build HPC-GAP if set to yes'
-    required: false
-    default: 'no'
-  ABI:
-    description: 'set to 32 to use 32bit build flags for the package'
+    default: 'gap-system/gap'
+  configflags:
+    description: 'Arguments to pass to the GAP configure script (e.g. --enable-debug)'
     required: false
     default: ''
-  CONFIGFLAGS:
-    description: 'arguments to pass to the GAP configure script (e.g. --enable-debug)'
+  gap-pkgs-to-build:
+    description: 'A space-separated list of the GAP packages to build'
     required: false
-    default: ''
-  GAP_BOOTSTRAP:
-    description: 'make bootstrap-pkg-? (i.e. full/minimal)'
-    required: false
-    default: 'full'
-env:
-  CHERE_INVOKING: 1
+    default: 'io json profiling'
+    deprecationMessage: 'This input will possibly be removed in the future, so do not rely on it too much'
+
 runs:
   using: "composite"
   steps:
-      - name: "Install dependencies"
-        run: |
-          echo "Installing dependencies"
-          if [ "${{runner.os}}" = "Linux" ]; then
-            if [ "${{inputs.ABI}}" = "32" ]; then
-              sudo dpkg --add-architecture i386
-              sudo apt-get update
-              packages=(
-                libgmp-dev:i386
-                libreadline-dev:i386
-                zlib1g-dev:i386
-                gcc-multilib
-                g++-multilib
-              )
+  
+    - name: "Install dependencies"
+      shell: bash
+      if: ${{ runner.os != 'Windows' }}
+      run: |
+        if [[ "${{ runner.os }}" == "Linux" ]] ; then
+          sudo apt-get install libgmp-dev libreadline-dev zlib1g-dev
+        elif [[ "${{ runner.os }}" = "macOS" ]] ; then
+          brew install zlib autoconf && brew link zlib --force
```

maybe also install `readline` to match Linux? And also the user of `--with-readline` later on...
```suggestion
          brew install zlib autoconf readline && brew link zlib --force
```